android/src/main/kotlin/io/chirp/chirpsdk/ChirpsdkPlugin.kt

```diff
@@ -51,22 +52,26 @@ class ChirpsdkPlugin(val activity: Activity) : MethodCallHandler {
 
   override fun onMethodCall(call: MethodCall, result: Result) {
 
+    //TODO: I think a switch->case is a better option here instead of multiple if/else
     if (call.method == "init") {
-      val arguments = call.arguments as java.util.HashMap<String, String>
+      val arguments = call.arguments as java.util.HashMap<*, *>
       val appKey = arguments["key"] as String
       val appSecret = arguments["secret"] as String
-      chirpConnect = ChirpConnect(activity, appKey, appSecret)
+      chirpSDK = ChirpSDK(activity, appKey, appSecret)
     }
     else if (call.method == "version") {
-      result.success(chirpConnect.version)
+      result.success(chirpSDK.version)
     }
     else if (call.method == "setConfig") {
-      var config: String = call.arguments as String
-      val error: ChirpError = chirpConnect.setConfig(config)
+      val config: String = call.arguments as String
+      val error: ChirpError = chirpSDK.setConfig(config)
       if (error.code > 0) {
+        //TODO: How do you know that errorHandler comes from this specific method call?
```

Comment:
Agree. This was done for a simple approach, but as you say, it makes error handling harder for the user. The same principle should also be applied to the react-native wrapper.